src/core/dense_set.cc

```diff
@@ -0,0 +1,556 @@
+// Copyright 2022, Roman Gershman.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/dense_set.h"
+
+#include <absl/numeric/bits.h>
+
+#include <stack>
+#include <type_traits>
+#include <vector>
+
+#include "glog/logging.h"
+
+extern "C" {
+#include "redis/zmalloc.h"
+}
+
+namespace dfly {
+using namespace std;
+
+constexpr size_t kMinSizeShift = 2;
+constexpr size_t kMinSize = 1 << kMinSizeShift;
+constexpr bool kAllowDisplacements = true;
+constexpr bool kFastScan = true;
+
+DenseSet::BucketIterator& DenseSet::BucketIterator::operator++() {
+  if (curr_) {
+    curr_ = curr_->Next();
+  }
+
+  return *this;
+}
+
+DenseSet::BucketConstIterator& DenseSet::BucketConstIterator::operator++() {
+  if (curr_) {
+    curr_ = curr_->Next();
+  }
+
+  return *this;
+}
+
+bool DenseSet::IsEmpty(DenseSet::DenseSetList it) const {
+  return it->IsEmpty();
+}
+
+int DenseSet::PushFront(DenseSet::DenseSetList it, void* data) {
```

Comment:
please make the return types consistent (also document their meaning in the header file).

here, for example, it should return `size_t` and not int.